instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/dynamodb.py

```diff
@@ -354,6 +374,12 @@ def __init__(self, call_context: _AwsSdkCallContext):
     def extract_attributes(self, attributes: _AttributeMapT):
         attributes[SpanAttributes.DB_SYSTEM] = DbSystemValues.DYNAMODB.value
         attributes[SpanAttributes.DB_OPERATION] = self._call_context.operation
+        attributes[SpanAttributes.DB_STATEMENT] = str(
```

Comment:
i wonder if capturing the `db.statement` for DynamoDB is really applicable in terms of semantic conventions, at least in the way how it is captured here.
right now this would capture (more or less) a serialized JSON string consisting of all the parameters that were passed to the DynamoDb SDK function (e.g. client.get_item(<param>)). This doesn't seem to look like defined format and backends will probably have a hard time interpreting it.

The instrumentation is also already capturing some of the parameters (e.g. TableName, ConsistentRead, ConsumedCapacity, ...) in other span attributes. So there will probably be some redundant/unrelated parts in the `db.statement`.
Additionally, boto3 provides at least two ways to call certain APIs. E.g. for the `get_item` (and most other) functions there exists a [client](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.get_item) API and a convenience [table](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Table.get_item) API. They are mostly the sames but slightly differ in e.g. how the `Key` parameter needs to be passed. For a captured unsanitized `db.statement` attribute this would probably mean that the statement would differ depending on which API is used.

Furthermore, there exists also the [PartiQL](https://partiql.org/) query language that is supported by certain DynamoDB APIs e.g. the [execute_statement](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.execute_statement). Currently the instrumentation doesn't capture any additional span attributes for any of these PartiQL specific APIs, but it might lead to inconsistencies in case support for it is added in the future, where then one API captures the `db.statement` as serialized JSON and another API captures PartiQL query string.